From 9797753775e7641c49fb253a47c52c2fad85439a Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 9 Apr 2013 15:49:21 +0300
Subject: [PATCH 05/10] conf ST: restore the operation

m0d survives unsupported opcode in "no confd" test case.
Make sure it's dead.

Remove noise.
---
 m0t1fs/linux_kernel/st/st | 28 ++++++++++++----------------
 1 file changed, 12 insertions(+), 16 deletions(-)

diff --git a/m0t1fs/linux_kernel/st/st b/m0t1fs/linux_kernel/st/st
index 8e78488..518ba69 100755
--- a/m0t1fs/linux_kernel/st/st
+++ b/m0t1fs/linux_kernel/st/st
@@ -1,6 +1,5 @@
 #!/bin/bash
-set -e
-set -u
+set -eu
 
 ## CAUTION: This path will be removed by superuser.
 SANDBOX_DIR=${SANDBOX_DIR:-~devvm/_m0-sandbox}
@@ -117,7 +116,8 @@ services_start() {
 
   ("prof-ac", {1| ("fs-1")}),
   ("fs-1", {2| ((11, 22),
-                [12: "-r", "-p", "-T", "linux", "-D", "db", "-S", "stobs", "-A", "addb-stobs", "-w", "3"],
+                [12: "-r", "-p", "-T", "linux", "-D", "db", "-S", "stobs",
+                     "-A", "addb-stobs", "-w", "3"],
                 [1: "ios-1"])}),
   ("ios-1", {3| (2, [1: "lnet:$IOS1_ENDPOINT"], "node-1")}),
   ("node-1", {4| (163840, 0, 0, 16, 0, [0], [0])})]
@@ -139,7 +139,6 @@ EOF`"
 
     echo "--- `date` ---" >>$SANDBOX_DIR/m0d.log
     cd $SANDBOX_DIR
-    echo $M0_CORE_DIR/mero/m0d $OPTIONS
     ($M0_CORE_DIR/mero/m0d $OPTIONS >>$SANDBOX_DIR/m0d.log 2>&1 &)
 
     sleep 1
@@ -151,7 +150,8 @@ EOF`"
 }
 
 services_stop() {
-    killall -q lt-m0d && sleep 5
+    killall -q lt-m0d && sleep 4
+    killall -q -KILL lt-m0d && sleep 1 || true
 }
 
 _mount() {
@@ -200,10 +200,8 @@ _init() {
 }
 
 _fini() {
-    ps ax | grep m0d
     services_stop
     sandbox_fini
-    lsmod | grep m0mero
     modules_remove
     exit ${1:-$?}
 }
@@ -259,30 +257,28 @@ for c in garbage premature_eof duplicated_ids no_fs_data bad_fs_params; do
     _mount local $c 2>/dev/null && error 'Failed to fail'
 done
 
-say "  Test: no network"
+say '  Test: no network'
 services_stop
 _mount net 2>/dev/null && error 'Failed to fail'
 
-say "  Test: no confd"
+say '  Test: no confd'
 services_start noconfd
 _mount net 2>/dev/null && error 'Failed to fail'
-killall -q -9 lt-m0d || error 'm0d was expected to be dead'
 
+# Restart services in order for confd to appear.
+services_stop
 services_start
 
 ## -------------------------------------------------------------------
-say "Suite: successes"
+say 'Suite: successes'
 
 echo 8 >/proc/sys/kernel/printk  # Print kernel messages to the console.
 
-###!!!!!!! Please fix the following testing and enable them
-_fini ; return
-
-say "  Test: m0t1fs local_conf"
+say '  Test: m0t1fs local_conf'
 _mount local || _fini $?
 umount $SANDBOX_DIR/mnt
 
-say "  Test: m0t1fs confd"
+say '  Test: m0t1fs confd'
 _mount net || _fini $?
 umount $SANDBOX_DIR/mnt
 
-- 
1.8.3.2

